File: 0801-0900/0866-Prime Palindrome/0866-Prime Palindrome.cpp
class Solution {
public:
    int primePalindrome(int N) {
        if (8 <= N && N <= 11) {
            return 11;
        }
        for (int i = 1; i < 100000; ++i) {
            string left = to_string(i), right(left.rbegin(), left.rend());
            int num = stoi(left + right.substr(1));
            if (num >= N && isPrime(num)) {
                return num;
            }
        }
        return -1;
    }
    
private:
    bool isPrime(int num) {
        if (num < 2 || num % 2 == 0) {
            return num == 2;
        }
        for (int i = 3; i * i <= num; i += 2) {
            if (num % i == 0) {
                return false;
            }
        }
            
        return true;
    }
};

File: 0801-0900/0866-Prime Palindrome/0866-Prime Palindrome.py
class Solution:
    def primePalindrome(self, N: int) -> int:
        if 8 <= N <= 11:
            return 11
        def isPrime(num):
            if num < 2 or num % 2 == 0:
                return num == 2
            return all(num % i for i in range(3, int(num**0.5) + 1, 2))
        
        for i in range(1, 100000):
            left = str(i)
            right = left[-2::-1]
            num = int(left + right)
            if num >= N and isPrime(num):
                return num
        return -1
